### DISCUSSION TITLE
Create functionality similar to Laravel Idea

### DISCUSSION THREAD

**Discussion by kirilldakhniuk** [EmranMR/Laravel-Nova-Extension#22] [Ideas]
I'm wondering are you thinking about implementing a full Laravel support? I guess it's a lot of work. But I didn't see any roadmap, so I'm asking here :) I really like how Laravel Idea works on PHPStorm and bringing some key features in Nova would be awesome. Of course Laravel Idea is a paid plugin. 

I even thought to create some of functionality myself. But I need to learn a lot first. So yeah, if you have roadmap or something please share :)

**Comment by EmranMR** [comment]
Hey @KirillDakhnyuk, yeah, that was the exact intention. However, as you mentioned, LarvelIDEA is a fully paid version, and as a result, it is in more active development. Also, the `tree-sitter-blade` turned out to be a bit more challenging than I thought.

Some of the stuff I had in mind is already out as an isolated extension, such as [Artisan](https://extensions.panic.com/extensions/teriyaki/teriyaki.LaravelArtisan/) and [pint](https://extensions.panic.com/extensions/kapalla/kapalla.pint/).

Surprisingly, I have only ever used Laravel IDEA briefly. (I just don't like how bulky PHPStorm is.) However, I know for a fact that the Laravel IDEA extension is the best in class when it comes to Laravel.

As you mentioned, it will be a lot of work to implement it all, considering that Laravel IDEA has been in development for many years. But we can surely get there in baby steps!

If you are experienced with IDEA, I would really appreciate it if you could drop a list of features here, and we can scramble a roadmap together. 😊

I think Nova has a solid API, so I am sure there are plenty of features that can already be implemented. But, as you said, the best way is to come up with the full list of features, prioritize them, and create a roadmap.

**Comment by kirilldakhniuk** [comment]
There's a lot useful features in the Laravel Idea, but I'm not a hard-core user so here's a list of features which I'm really missing in Nova.

- autocompletion inside a helper functions like `config()`, `route()`, etc...
- go to method, go to file, go to route, etc...
- model properties autocompletion
- defining relationships
- validation rules helper
- autocompletion inside a `whereHas` , etc...

Potentially I could try to help you, but I need to figure out some stuff first :)
